code/game/objects/items/stacks/stack.dm

```diff
@@ -22,7 +22,11 @@
 /obj/item/stack/New(var/loc, var/amount=null)
 	..()
 	if (amount)
-		src.amount = amount
+		if(amount<=max_amount)
+			src.amount = amount
```

Comment:
new is called with a variable called amount. src is used to set src amount to the amount argument. Isn't this an actual need for it?